### PR TITLE
shfmt: fix shell formatting

### PR DESCRIPTION
**Pull request by alban**
While working on the proteus branch (https://github.com/weaveworks/scope/issues/2433), I noticed the following errors:

```
./tools/lint
bin/release: run shfmt -i 4 -w bin/release
extras/dialer/time-scope-probe-cgroup:31:18: arithmetic expressions must consist of names and numbers
extras/dialer/time-scope-probe-cgroup: run shfmt -i 4 -w extras/dialer/time-scope-probe-cgroup
extras/dialer/time-scope-probe-proc:28:18: arithmetic expressions must consist of names and numbers
extras/dialer/time-scope-probe-proc: run shfmt -i 4 -w extras/dialer/time-scope-probe-proc
extras/in_parallel.sh:29:16: arithmetic expressions must consist of names and numbers
extras/in_parallel.sh: run shfmt -i 4 -w extras/in_parallel.sh
scope: run shfmt -i 4 -w scope
tools/dependencies/list_os_images.sh:11:3: $ must be escaped or followed by a literal
tools/dependencies/list_os_images.sh: run shfmt -i 4 -w tools/dependencies/list_os_images.sh
tools/integration/assert.sh: run shfmt -i 4 -w tools/integration/assert.sh
tools/provisioning/setup.sh:301:3: $ must be escaped or followed by a literal
tools/provisioning/setup.sh: run shfmt -i 4 -w tools/provisioning/setup.sh
tools/test:115:22: arithmetic expressions must consist of names and numbers
tools/test: run shfmt -i 4 -w tools/test
Makefile:132: recipe for target 'lint' failed
```
visible on https://circleci.com/gh/kinvolk/scope/980

This PR fixes that:
- in extras/
- in tools/ (updated with git-subtree)

### CODE REVIEW

File: extras/dialer/time-scope-probe-cgroup [hunk 1]
```diff
@@ -28,7 +28,7 @@ TIME_K2=$(awk '/system/ {print $2}' "${cpuacct_stat}")
 TIME_T2=$((TIME_U2 + TIME_K2))
 
 echo "utime;stime;total_time"
-for ((i = 0; i < "$interval_num"; i++)); do
+for ((i = 0; i < interval_num; i++)); do
```
Comment (2opremio) [inline extras/dialer/time-scope-probe-cgroup]: This can't be right

File: extras/dialer/time-scope-probe-proc [hunk 2]
```diff
@@ -25,7 +25,7 @@ TIME_K2=$(gawk '{print $15"*10"}' <"/proc/$(pidof scope-probe)/stat" | bc)
 TIME_T2=$((TIME_U2 + TIME_K2))
 
 echo "utime;stime;total_time"
-for ((i = 0; i < "$interval_num"; i++)); do
+for ((i = 0; i < interval_num; i++)); do
```
Comment (2opremio) [inline extras/dialer/time-scope-probe-proc]: Same here

File: extras/dialer/time-scope-probe-cgroup (hunk 1 above)
Comment (alban) [inline extras/dialer/time-scope-probe-cgroup]: Apparently, with the double-parenthesis construct, no need for the `$` prefix on variables, or quotes. @schu told me so today.

http://tldp.org/LDP/abs/html/dblparens.html

Comment (2opremio) [inline extras/dialer/time-scope-probe-cgroup]: Nice, I didn't know

**Comment by 2opremio** [comment]
Let's hold off from merging until that is resolved.

**Comment by 2opremio** [comment]
(Otherwise LGTM)

**Comment by 2opremio** [comment]
@alban I am taking care of the failure at https://github.com/weaveworks/scope/pull/2531

**Comment by 2opremio** [comment]
Targeting the release 1.4.0 branch

**Comment by 2opremio** [comment]
I ended up removing this change from the release branch since it was still failing. Mind recreating the PR against master and cherry-picking https://github.com/weaveworks/scope/commit/dd2c503748618e752700e1331d675e8b33f24a8e

Sorry about the mess @alban

**Pull request merged**